### PR TITLE
Admin worklogs

### PR DESCRIPTION
**Pull request by bisheshbh**
Added feedback feature , able to delete updates by admin , edit user by admin , change password , department by user , search by department and date added.

### CODE REVIEW

File: src/admin/controllers/admin.controller.ts [hunk 1]
```diff
@@ -1,21 +1,110 @@
...
+            await adminModel.createFeedback(req.body.comment , req.body.created_date , taskId, userId)
+            return res.redirect('/admin/worklogs')
+        } catch (error) {
+            return res.render('admin/admin-feedback', {errors:[{msg:"Something went wrong!"}] , task, createdDate, taskId});
+        }
+         
+    }
+
+    getUpdateUser : RequestHandler = async(req:express.Request, res:express.Response) => {
+        const userId = +req.params.id;
```
Comment (chandan-introcept) [inline src/admin/controllers/admin.controller.ts:70]: why +req.params.id ?? any reason

File: src/migrations/create-table.ts [hunk 2]
```diff
@@ -15,36 +16,44 @@ class CreateTable {
             username VARCHAR(25),
             email VARCHAR(150),
             password VARCHAR(100),
-            date_of_birth DATE,
+            date_of_birth VARCHAR(25),
```
Comment (chandan-introcept) [inline src/migrations/create-table.ts:19]: why date_of_birth changed to varchar?? this should be date type

File: src/users/models/users.models.ts [hunk 3]
```diff
@@ -1,55 +1,105 @@
...
+        SELECT username , email , date_of_birth, address, 
+        department_name FROM user INNER JOIN department 
+        ON user.department_id=department.id WHERE user.id=${id}
+        `
+        const result:any = await db.run(sql);
+        return result;
+    }
     
-    findAll = async (email:string) : Promise<Object|[]> =>{
+    findAll = async () : Promise<Object|[]> =>{
```
Comment (chandan-introcept) [inline src/users/models/users.models.ts:47]: getAll() will be meaningful

File: src/worklogs/models/worklogs.model.ts [hunk 4]
```diff
@@ -1,84 +1,156 @@
...
+        const sql = 
+        `DELETE FROM feedback WHERE task_id=${task_id}`
+        try {
+            await db.run(sql);
+        } catch (error) {
+            throw error
+        }
+    }
+
+    deleteTask = async(task_id:number) => {
```
Comment (chandan-introcept) [inline src/worklogs/models/worklogs.model.ts:143]: we should check if task exists or not before  deleting it

**Review by chandan-introcept — CHANGES_REQUESTED** [review]

File: src/admin/controllers/admin.controller.ts (hunk 1 above)
Comment (bisheshbh) [reply src/admin/controllers/admin.controller.ts:72]: for converting the id parameter(string) to number .

File: src/migrations/create-table.ts (hunk 2 above)
Comment (bisheshbh) [reply src/migrations/create-table.ts]: changed back to date.

**Pull request merged**